test/libcommonserver/io/testchecksetgetrights.cpp

```diff
@@ -0,0 +1,298 @@
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "testio.h"
+#include "libcommonserver/utility/utility.h"
+
+#include <filesystem>
+
+using namespace CppUnit;
+
+namespace KDC {
+
+static struct rightsSet {
+        bool read;
+        bool write;
+        bool execute;
+};
+
+static void rightsSetFromInt(int rights, rightsSet& rightsSet) {
+    rightsSet.read = rights & 4;
+    rightsSet.write = rights & 2;
+    rightsSet.execute = rights & 1;
+}
+
+static int rightsSetToInt(rightsSet rightsSet) {
+    int rights = 0;
+    if (rightsSet.read) {
+        rights += 4;
+    }
+    if (rightsSet.write) {
+        rights += 2;
+    }
+    if (rightsSet.execute) {
+        rights += 1;
+    }
+
+    return rights;
+}
+
+void TestIo::testCheckSetAndGetRights() {
+    std::cout << "TestIo::testCheckSetAndGetRights: start" << std::endl;
+
+#ifdef _WIN32
+    std::cout << "TestIo::testCheckSetAndGetRights: init trustee" << std::endl;
+    CPPUNIT_ASSERT(Utility::init());  // Initialize the utility library, needed to access/change the permissions on Windows
+#endif
+
+    const TemporaryDirectory temporaryDirectory;
+
+    // Test if the rights are correctly set and get on a directory
+    {
+        const SyncPath path = temporaryDirectory.path / "changePerm";
+
+        IoError ioError = IoErrorUnknown;
+        CPPUNIT_ASSERT(IoHelper::createDirectory(path, ioError));
+        CPPUNIT_ASSERT(ioError == IoErrorSuccess);
+
+        bool isReadable = false;
+        bool isWritable = false;
+        bool isExecutable = false;
+        bool exists = false;
+
+        rightsSet rightsSet = {false, false, false};
+
+        // For a directory
+
+        /* Test all the possible rights and the all the possible order of rights modification. ie:
+         *  | READ | WRITE | EXECUTE | |
+         *  |  0   |   0   |    0    | v
+         *  |  0   |   0   |    1    | |
+         *  |  0   |   0   |    0    | v
+         *  |  0   |   1   |    0    | |
+         *  |  0   |   0   |    0    | v
+         *  |  0   |   1   |    1    | |
+         *  |  0   |   0   |    0    | v
+         *  |  1   |   0   |    0    | |
+         *  |  0   |   0   |    0    | v
+         *  |  1   |   0   |    1    | |
+         *  |  0   |   0   |    0    | v
+         *  |  1   |   1   |    0    | |
+         *  |  0   |   0   |    0    | v
+         *  |  1   |   1   |    1    | |
+         *  |  0   |   0   |    0    | v
+         *  ...
+         */
+
+        for (int baseRigths = 0; baseRigths < 7;
+             baseRigths++) {  // Test all the possible rights and the all the possible order of rights modification
+            for (int targetRigths = baseRigths + 1; targetRigths < 8; targetRigths++) {
+                rightsSetFromInt(baseRigths, rightsSet);
```

Comment:
You may implement a constructor taking an `int` as argument and then use a syntax like this:

```
rightSet = RightsSet(baseRights);
```

Reply:
I implemented your suggestion in commit: https://github.com/Infomaniak/desktop-kDrive/pull/68/commits/c435db833e99dab58407ed5f7386c12499fd6d43